processing/src/main/java/io/druid/query/timeboundary/TimeBoundaryQueryRunnerFactory.java

```diff
@@ -81,6 +92,38 @@ public TimeBoundaryQueryRunner(Segment segment)
       this.adapter = segment.asStorageAdapter();
     }
 
+    private final Function<Cursor, Result<DateTime>> skipToFirstMatching = new Function<Cursor, Result<DateTime>>()
```

Comment:
Please declare this above the constructor (`private final Function<Cursor, Result<DateTime>> skipToFirstMatching`) and then initialize it in the constructor.